File: app/src/main/java/br/com/douglaspac/qualcombustivel/MainActivity.java
package br.com.douglaspac.qualcombustivel;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import com.shawnlin.numberpicker.NumberPicker;

public class MainActivity extends AppCompatActivity
{
    NumberPicker mPickEtanol1;
    NumberPicker mPickEtanol2;
    NumberPicker mPickEtanol3;
    NumberPicker mPickEtanol4;

    NumberPicker mPickGasolina1;
    NumberPicker mPickGasolina2;
    NumberPicker mPickGasolina3;
    NumberPicker mPickGasolina4;

    Button mBtnCalculate;
    @Override
    protected void onCreate(Bundle savedInstanceState)
    {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Float gas = SaveSharedPreferences.getGas(this);
        Float etanol = SaveSharedPreferences.getEtanol(this);

        String gasStr = String.format("%.3f", gas);
        String etanolStr = String.format("%.3f", etanol);

        mPickEtanol1 = (NumberPicker) findViewById(R.id.nmb_pick_etanol_1);
        mPickEtanol2 = (NumberPicker) findViewById(R.id.nmb_pick_etanol_2);
        mPickEtanol3 = (NumberPicker) findViewById(R.id.nmb_pick_etanol_3);
        mPickEtanol4 = (NumberPicker) findViewById(R.id.nmb_pick_etanol_4);

        mPickEtanol1.setValue(etanolStr.charAt(0) - '0');
        mPickEtanol2.setValue(etanolStr.charAt(2) - '0');
        mPickEtanol3.setValue(etanolStr.charAt(3) - '0');
        mPickEtanol4.setValue(etanolStr.charAt(4) - '0');

        mPickGasolina1 = (NumberPicker) findViewById(R.id.nmb_pick_gasolina_1);
        mPickGasolina2 = (NumberPicker) findViewById(R.id.nmb_pick_gasolina_2);
        mPickGasolina3 = (NumberPicker) findViewById(R.id.nmb_pick_gasolina_3);
        mPickGasolina4 = (NumberPicker) findViewById(R.id.nmb_pick_gasolina_4);

        mPickGasolina1.setValue(gasStr.charAt(0) - '0');
        mPickGasolina2.setValue(gasStr.charAt(2) - '0');
        mPickGasolina3.setValue(gasStr.charAt(3) - '0');
        mPickGasolina4.setValue(gasStr.charAt(4) - '0');

        mBtnCalculate = (Button) findViewById(R.id.button);

        mBtnCalculate.setOnClickListener(new View.OnClickListener()
        {
            @Override
            public void onClick(View v)
            {
                Float etanol = Float.valueOf(String.valueOf(mPickEtanol1.getValue())
                                                + "."
                                                + String.valueOf(mPickEtanol2.getValue())
                                                + String.valueOf(mPickEtanol3.getValue())
                                                + String.valueOf(mPickEtanol4.getValue()));
                Float gas = Float.valueOf(String.valueOf(mPickGasolina1.getValue())
                                                + "."
                                                + String.valueOf(mPickGasolina2.getValue())
                                                + String.valueOf(mPickGasolina3.getValue())
                                                + String.valueOf(mPickGasolina4.getValue()));

                SaveSharedPreferences.setEtanol(MainActivity.this, etanol);
                SaveSharedPreferences.setGas(MainActivity.this, gas);

                if(etanol / gas < 0.7)
                {
                    mBtnCalculate.setText(getString(R.string.go_alcool));
                }
                else
                {
                    mBtnCalculate.setText(getString(R.string.go_gas));
                }
            }
        });
    }
}

File: app/src/main/java/br/com/douglaspac/qualcombustivel/MainActivity.kt
package br.com.douglaspac.qualcombustivel.kotlin

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.widget.Button
import br.com.douglaspac.qualcombustivel.R
import kotlinx.android.synthetic.main.activity_main.*
import java.util.*


class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val gas = SaveSharedPreferences.getGas(this)
        val etanol = SaveSharedPreferences.getEtanol(this)

        val gasStr = String.format("%.3f", gas)
        val etanolStr = String.format("%.3f", etanol)

        val mPickEtanol1 = this.nmb_pick_etanol_1
        val mPickEtanol2 = this.nmb_pick_etanol_2
        val mPickEtanol3 = this.nmb_pick_etanol_3
        val mPickEtanol4 = this.nmb_pick_etanol_4

        mPickEtanol1.value = etanolStr[0] - '0'
        mPickEtanol2.value = etanolStr[2] - '0'
        mPickEtanol3.value = etanolStr[3] - '0'
        mPickEtanol4.value = etanolStr[4] - '0'

        val mPickGasolina1 = this.nmb_pick_gasolina_1
        val mPickGasolina2 = this.nmb_pick_gasolina_2
        val mPickGasolina3 = this.nmb_pick_gasolina_3
        val mPickGasolina4 = this.nmb_pick_gasolina_4

        mPickGasolina1.value = gasStr[0] - '0'
        mPickGasolina2.value = gasStr[2] - '0'
        mPickGasolina3.value = gasStr[3] - '0'
        mPickGasolina4.value = gasStr[4] - '0'

        val mBtnCalculate = findViewById(R.id.button) as Button

        mBtnCalculate.setOnClickListener {
            val etanol = java.lang.Float.valueOf(mPickEtanol1.value.toString()
                    + "."
                    + mPickEtanol2.value.toString()
                    + mPickEtanol3.value.toString()
                    + mPickEtanol4.value.toString())
            val gas = java.lang.Float.valueOf(mPickGasolina1.value.toString()
                    + "."
                    + mPickGasolina2.value.toString()
                    + mPickGasolina3.value.toString()
                    + mPickGasolina4.value.toString())

            SaveSharedPreferences.setEtanol(this@MainActivity, etanol)
            SaveSharedPreferences.setGas(this@MainActivity, gas)

            if (etanol!! / gas!! < 0.7) {
                mBtnCalculate.text = getString(R.string.go_alcool)
            } else {
                mBtnCalculate.text = getString(R.string.go_gas)
            }
        }
    }
}
